README.md

```diff
@@ -6,29 +6,98 @@ Based on https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-
+- **args** : corresponding to third parameter of `register_rest_route` function : [doc](https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-custom-endpoints/#arguments)
+
+```php
+
+ class QuestionApiService extends AbstractApiService
+{
+    /**
+     * @WPApiEndpoint(
+     *     namespace = "questions",
+     *     version = "v1",
+     *     url = "/questions-by-user",
+     *     args = {
+    *       "methods": "GET",
+    *       "args": {
+    *           "page": {
+    *               "default": 1,
+    *               "sanitize_callback": "convertToInt"
+    *           }, 
+    *           "limit": {
+    *               "default": 1,
+    *               "sanitize_callback": "convertToInt"
+    *           }
+    *       },
+    *       "permission_callback": "canFetchQuestions"
+    *     }
+     * )
+     */
+    public function questionsByUser(WP_REST_Request $request)
+    {
+        $page = $request->get_param('page');
+        
+        /* Processing get questions */
+        
+        // To handle error we can return WP_Error instance
+        return WP_Error('error_code', 'Error message', ['status' => 400]);
+    
+        // Or success response
+        return new WP_REST_Response([
+            'questions' => [],
+        ]);
+    }
+    
+    public function canFetchQuestions(WP_REST_Request $request) 
+    {
+        $user = wp_get_current_user();
+        
+        /* processing verification */
+        
+        return true;
+    }
+    
+    public function convertToInt($param, $request, $key)
+    {
+        /* Processing sanitize method */
+        return (int) $param;
+    }
+}
+```
+
+The namespace could be define globaly.
+For example to create this endpoint `https://domain.test/wp-json/questions/v1/delete-question/1` 
 
 ```php
 /**
- * @WPApiEndpoint(
- *     namespace = "questions",
- *     version = "v1",
- *     url = "/questions-by-user",
- *     method = "GET",
- *     args = {"page": {"default": 1}, "limit": {"default": 1}}
+ * @WPApiNamespace(
+ *     namespace="questions"
  * )
  */
-public function questionsByUser(WP_REST_Request $request)
+class QuestionApiService extends AbstractApiService
 {
-    $userId = $request->get_param('userId');
+/**
+     * @WPApiEndpoint(
+     *     url = "/delete-question/(?P<id>\d+)",
+     *     args = {
+     *      "methods": "DELETE",
+     *      "permission_callback": "canDeleteQuestion"
+     *     }
+     * )
+     */
+    public function deleteQuestion(WP_REST_Request $request)
+    {
+        /* Processing deletion */
+    }
     
-    /* ... */
-
-    return new WP_REST_Response([
-        'questions' => [],
-    ]);
+    public function canDeleteQuestion()
+    {}
 }
 ```
+
+### Important notice
+
+For `sanitize_callback`, `validate_callback` and `permission_callback`, we are looking for an instance method on the same class of the annotation.
```

Comment:
```suggestion
Because we cannot reference `$this` from within an annotation, for any callback function specified inside a @WPApiEndpoint annotation, be it `sanitize_callback`, `validate_callback`, `permission_callback` and so on, the provided callback name (for example `myFunction`), is first looked for on the same class instance than the method carrying out the annotation. If not found there, it's then looked for on the global namespace (like a function name)
```